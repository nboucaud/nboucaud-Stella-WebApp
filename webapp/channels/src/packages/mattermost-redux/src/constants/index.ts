// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import Bot from './bot';
import Emoji from './emoji';
import Files from './files';
import General from './general';
import Permissions from './permissions';
import Plugins from './plugins';
import Posts from './posts';
import Preferences from './preferences';
import RequestStatus from './request_status';
import Roles from './roles';
import Stats from './stats';
import Teams from './teams';
import Threads from './threads';
import Users from './users';
import WebsocketEvents from './websocket';

export {
    Bot,
    Emoji,
    Files,
    General,
    Permissions,
    Plugins,
    Posts,
    Preferences,
    RequestStatus,
    Roles,
    Stats,
    Teams,
    Threads,
    Users,
    WebsocketEvents,
};
