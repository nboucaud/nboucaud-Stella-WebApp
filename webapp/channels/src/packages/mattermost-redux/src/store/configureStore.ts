// Import necessary modules
import {composeWithDevTools} from '@redux-devtools/extension';
import {
    applyMiddleware,
    legacy_createStore,
} from 'redux';
import type {
    Reducer,
    Store,
} from 'redux';
import thunk from 'redux-thunk';

import type {GlobalState} from '@mattermost/types/store';

import {createReducer} from './helpers';
import initialState from './initial_state';
import reducerRegistry from './reducer_registry';
import modalReducer from '../reducers/entities/modalReducer';
import serviceReducers from '../reducers';
import tabReducer from 'mattermost-redux/reducers/entities/tabReducer';
import urlManagerReducer from '../../../../components/browser_view/browser/browser-state'; // Import your new reducer

/**
 * Configures and constructs the redux store. Accepts the following parameters:
 * preloadedState - Any preloaded state to be applied to the store after it is initially configured.
 * appReducers - An object containing any app-specific reducer functions that the client needs.
 * getAppReducers - A function that returns the appReducer as defined above. Only used in development to enable hot reloading.
 */
export default function configureStore<S extends GlobalState>({
    appReducers,
    getAppReducers,
    preloadedState,
}: {
    appReducers: Record<string, Reducer>;
    getAppReducers: () => Record<string, Reducer>;
    preloadedState: Partial<S>;
}): Store {
    const baseState = {
        ...initialState,
        ...preloadedState,
    };

    const composeEnhancers = composeWithDevTools({
        shouldHotReload: false,
        trace: true,
        traceLimit: 25,
        autoPause: true,
    });

    const middleware = applyMiddleware(
        thunk.withExtraArgument({loaders: {}}),
    );

    const enhancers = composeEnhancers(middleware);

    // Add the urlManagerReducer to the appReducers
    const baseReducer = createReducer({
        ...serviceReducers,
        modal: modalReducer, 
        tabs: tabReducer,
        urlManager: urlManagerReducer, // Add your URL manager reducer here
        ...appReducers,
    });

    const store = legacy_createStore(
        baseReducer,
        baseState,
        enhancers
    );

    reducerRegistry.setChangeListener((reducers: Record<string, Reducer>) => {
        store.replaceReducer(createReducer(reducers, serviceReducers, appReducers));
    });

    if (module.hot) {
        // Enable Webpack hot module replacement for reducers
        module.hot.accept(() => {
            const registryReducers = reducerRegistry.getReducers();
            const nextServiceReducers = require('../reducers').default; // eslint-disable-line global-require
            const nextAppReducers = getAppReducers();

            // Ensure registryReducers comes first so that stored service/app reducers are replaced by the new ones
            store.replaceReducer(createReducer(registryReducers, nextServiceReducers, nextAppReducers));
        });
    }

    return store;
}
