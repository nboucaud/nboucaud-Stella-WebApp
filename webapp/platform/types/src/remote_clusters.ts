// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type RemoteClusterInvite = {
    remote_id: string;
    remote_team_id: string;
    site_url: string;
    token: string;
};

export type RemoteClusterAcceptInvite = {
    name: string;
    display_name: string;
    invite: string;
    password: string;
}

export type RemoteClusterPatch = Pick<RemoteCluster, 'display_name'>

export function isRemoteClusterPatch(x: Partial<RemoteClusterPatch>): x is RemoteClusterPatch {
    return x.display_name !== undefined && x.display_name !== '';
}

export type RemoteCluster = {
    remote_id: string;
    remote_team_id: string;
    name: string;
    display_name: string;
    site_url: string;
    create_at: number;
    last_ping_at: number;
    token: string;
    remote_token?: string;
    topics: string;
    creator_id: string;
    plugin_id: string;
    options: number;
}

export type RemoteClusterWithPassword = RemoteCluster & {
    password: string;
}
