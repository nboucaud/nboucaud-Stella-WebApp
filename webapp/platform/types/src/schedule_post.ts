// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {Draft} from './drafts';

export type SchedulingInfo = {
    scheduled_at: number;
    processed_at?: number;
    error_code?: string;
}

export type ScheduledPost = Omit<Draft, 'delete_at'> & SchedulingInfo & {
    id: string;
}

export type ScheduledPostsState = {
    byTeamId: {[teamId: string]: ScheduledPost[]};
}
