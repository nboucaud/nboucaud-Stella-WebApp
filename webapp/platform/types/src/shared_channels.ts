// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type SharedChannel = {
    id: string;
    team_id: string;
    home: boolean;
    readonly: boolean;
    name: string;
    display_name: string;
    purpose: string;
    header: string;
    creator_id: string;
    create_at: number;
    update_at: number;

    /** Omitted when when {@link SharedChannel.home} is `false` */
    remote_id?: string;
};

export type SharedChannelRemote = {
    id: string;
    channel_id: string;
    creator_id: string;
    create_at: number;
    update_at: number;
    is_invite_accepted: boolean;
    is_invite_confirmed: boolean;
    remote_id: string;
    last_post_update_at: number;
    last_post_id: string;
    last_post_create_at: number;
    last_post_create_id: string;
}
