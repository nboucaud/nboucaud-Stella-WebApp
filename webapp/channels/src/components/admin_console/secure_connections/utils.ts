// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {LocationDescriptor} from 'history';
import {useEffect, useState} from 'react';
import {useDispatch} from 'react-redux';

import type {ClientError} from '@mattermost/client';
import type {RemoteCluster, RemoteClusterAcceptInvite, RemoteClusterPatch} from '@mattermost/types/remote_clusters';
import type {PartialExcept} from '@mattermost/types/utilities';

import {Client4} from 'mattermost-redux/client';

import {openModal} from 'actions/views/modals';

import {ModalIdentifiers} from 'utils/constants';
import {cleanUpUrlable} from 'utils/url';

import SecureConnectionAcceptInviteModal from './secure_connection_accept_invite_modal';
import SecureConnectionCreateInviteModal from './secure_connection_create_invite_modal';
import SecureConnectionDeleteModal from './secure_connection_delete_modal';

export const useRemoteClusters = () => {
    const [remoteClusters, setRemoteClusters] = useState<RemoteCluster[]>();
    const [loadingState, setLoadingState] = useState<boolean | ClientError>(true);

    const isLoading = loadingState === true;

    const fetch = async () => {
        setLoadingState(true);
        try {
            const data = await Client4.getRemoteClusters({excludePlugins: true});
            setRemoteClusters(data?.length ? data : undefined);
            setLoadingState(false);
        } catch (err) {
            setLoadingState(err);
        }
    };

    useEffect(() => {
        fetch();
    }, []);

    return [remoteClusters, {isLoading, fetch}] as const;
};

export const useRemoteClusterEdit = (remoteId: string | 'create', initRemoteCluster?: RemoteCluster) => {
    const editing = remoteId !== 'create';

    const [currentRemoteCluster, setCurrentRemoteCluster] = useState<RemoteCluster | undefined>(initRemoteCluster);
    const [patch, setPatch] = useState<Partial<RemoteClusterPatch>>({});
    const [loading, setLoading] = useState<boolean | ClientError>(editing && !currentRemoteCluster);
    const [saving, setSaving] = useState<boolean | ClientError>(false);

    const hasChanges = Object.keys(patch).length > 0;

    useEffect(() => {
        if (!editing) {
            return;
        }
        (async () => {
            try {
                const data = await Client4.getRemoteCluster(remoteId);
                setCurrentRemoteCluster(data);
                setLoading(false);
                setPatch({});
            } catch (err) {
                setSaving(err);
            }
        })();
    }, [remoteId]);

    const applyPatch = (patch: Partial<RemoteClusterPatch>) => {
        setPatch((current) => ({...current, ...patch}));
    };

    const save = async () => {
        if (currentRemoteCluster && hasChanges) {
            if (patch.display_name === undefined) {
                return;
            }
            setSaving(true);
            try {
                const data = await Client4.patchRemoteCluster(remoteId, patch);
                setCurrentRemoteCluster(data);
                setSaving(false);
                setPatch({});
            } catch (err) {
                setSaving(err);
            }
            setSaving(false);
        }
    };

    return [
        {...currentRemoteCluster, ...patch},
        {
            applyPatch,
            save,
            hasChanges,
            loading,
            saving,
            currentRemoteCluster,
            patch,
        },
    ] as const;
};

const chars = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz~_!@-#$^';
const makePassword = () => {
    return Array.from(window.crypto.getRandomValues(new Uint32Array(16))).
        map((n) => chars[n % chars.length]).
        join('');
};

export const useRemoteClusterCreate = () => {
    const dispatch = useDispatch();
    const [saving, setSaving] = useState<boolean | ClientError>(false);

    const promptCreate = (patch: RemoteClusterPatch) => {
        return new Promise<RemoteCluster | undefined>((resolve, reject) => {
            dispatch(openModal({
                modalId: ModalIdentifiers.SECURE_CONNECTION_CREATE_INVITE,
                dialogType: SecureConnectionCreateInviteModal,
                dialogProps: {
                    creating: true,
                    password: makePassword(),
                    onConfirm: async (password: string) => {
                        try {
                            setSaving(true);
                            const response = await Client4.createRemoteCluster({
                                ...patch,
                                name: cleanUpUrlable(patch.display_name),
                                password,
                            });
                            setSaving(false);

                            if (response) {
                                const {invite, remote_cluster: remoteCluster} = response;

                                resolve(remoteCluster);
                                return {remoteCluster, share: {invite, password}};
                            }
                        } catch (err) {
                            // handle create error
                            reject(err);
                        }
                        setSaving(false);
                        return undefined;
                    },
                },
            }));
        });
    };

    return {promptCreate, saving} as const;
};

export const useRemoteClusterCreateInvite = (remoteCluster: RemoteCluster) => {
    const dispatch = useDispatch();
    const [saving, setSaving] = useState<boolean | ClientError>(false);

    const promptCreateInvite = () => {
        return new Promise<RemoteCluster>((resolve, reject) => {
            dispatch(openModal({
                modalId: ModalIdentifiers.SECURE_CONNECTION_CREATE_INVITE,
                dialogType: SecureConnectionCreateInviteModal,
                dialogProps: {
                    password: makePassword(),
                    onConfirm: async (password: string) => {
                        try {
                            setSaving(true);
                            const invite = await Client4.generateInviteRemoteCluster(remoteCluster.remote_id, {password});
                            setSaving(false);
                            resolve(remoteCluster);
                            return {remoteCluster, share: {invite, password}};
                        } catch (err) {
                            // handle create error
                            reject(err);
                        }
                        setSaving(false);
                        return undefined;
                    },
                },
            }));
        });
    };

    return {promptCreateInvite, saving} as const;
};

export const useRemoteClusterAcceptInvite = () => {
    const dispatch = useDispatch();
    const [saving, setSaving] = useState<boolean | ClientError>(false);

    const promptAcceptInvite = () => {
        return new Promise<RemoteCluster>((resolve, reject) => {
            dispatch(openModal({
                modalId: ModalIdentifiers.SECURE_CONNECTION_CREATE_INVITE,
                dialogType: SecureConnectionAcceptInviteModal,
                dialogProps: {
                    onConfirm: async (acceptInvite: PartialExcept<RemoteClusterAcceptInvite, 'display_name' | 'invite' | 'password'>) => {
                        try {
                            setSaving(true);
                            const rc = await Client4.acceptInviteRemoteCluster({
                                ...acceptInvite,
                                name: cleanUpUrlable(acceptInvite.display_name),
                            });
                            setSaving(false);
                            resolve(rc);
                            return rc;
                        } catch (err) {
                            // handle create error
                            reject(err);
                            setSaving(err);
                            throw (err);
                        }
                    },
                },
            }));
        });
    };

    return {promptAcceptInvite, saving} as const;
};

export const useRemoteClusterDelete = (rc: RemoteCluster) => {
    const dispatch = useDispatch();
    const promptDelete = () => {
        return new Promise((resolve, reject) => {
            dispatch(openModal({
                modalId: ModalIdentifiers.SECURE_CONNECTION_DELETE,
                dialogType: SecureConnectionDeleteModal,
                dialogProps: {
                    displayName: rc.display_name,
                    onConfirm: () => Client4.deleteRemoteCluster(rc.remote_id).then(resolve, reject),
                },
            }));
        });
    };

    return {promptDelete} as const;
};

export const getEditLocation = (rc: RemoteCluster): LocationDescriptor<RemoteCluster> => {
    return {pathname: `/admin_console/environment/secure_connections/${rc.remote_id}`, state: rc};
};

export const getCreateLocation = (): LocationDescriptor<RemoteCluster> => {
    return {pathname: '/admin_console/environment/secure_connections/create'};
};

const SiteURLPendingPrefix = 'pending_';
export const isPending = (remoteCluster: RemoteCluster) => remoteCluster.site_url.startsWith(SiteURLPendingPrefix);
export const isConnected = (remoteCluster: RemoteCluster) => !isPending(remoteCluster);

export const isPendingState = <T extends boolean | unknown>(x: T) => x === true;
export const isErrorState = <T extends boolean | unknown>(x: T) => Boolean(typeof x !== 'boolean' && x);
