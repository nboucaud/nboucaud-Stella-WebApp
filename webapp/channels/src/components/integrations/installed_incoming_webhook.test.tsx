// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {render, screen} from '@testing-library/react';
import React from 'react';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';

import {wrapIntl} from '@mattermost/components/src/testUtils';
import type {IncomingWebhook} from '@mattermost/types/integrations';

import store from 'stores/redux_store';

import InstalledIncomingWebhook, {matchesFilter} from 'components/integrations/installed_incoming_webhook';

import {getHistory} from 'utils/browser_history';

describe('components/integrations/InstalledIncomingWebhook', () => {
    const incomingWebhook: IncomingWebhook = {
        id: '9w96t4nhbfdiij64wfqors4i1r',
        channel_id: '1jiw9kphbjrntfyrm7xpdcya4o',
        create_at: 1502455422406,
        delete_at: 0,
        description: 'build status',
        display_name: 'build',
        team_id: 'eatxocwc3bg9ffo9xyybnj4omr',
        update_at: 1502455422406,
        user_id: 'zaktnt8bpbgu8mb6ez9k64r7sa',
        username: 'username',
        icon_url: 'http://test/icon.png',
        channel_locked: false,
    };

    const teamId = 'testteamid';

    const ComponentWrapper = ({children}: {children: React.ReactElement<IncomingWebhook>}) => {
        return (
            <>
                {wrapIntl(
                    <Provider store={store}>
                        <Router history={getHistory()}>
                            {children}
                        </Router>
                    </Provider>,
                )}
            </>
        );
    };

    test('should match snapshot', () => {
        function emptyFunction() {} //eslint-disable-line no-empty-function

        const {asFragment} = render(
            <ComponentWrapper>
                <InstalledIncomingWebhook
                    key={1}
                    incomingWebhook={incomingWebhook}
                    onDelete={emptyFunction}
                    creator={{username: 'creator'}}
                    canChange={true}
                    team={{
                        id: teamId,
                        name: 'test',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        type: 'O',
                        display_name: 'name',
                        scheme_id: 'id',
                        allow_open_invite: false,
                        group_constrained: false,
                        description: '',
                        email: '',
                        company_name: '',
                        allowed_domains: '',
                        invite_id: '',
                    }}
                    channel={{
                        id: '1jiw9kphbjrntfyrm7xpdcya4o',
                        name: 'town-square',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        team_id: teamId,
                        type: 'O',
                        display_name: 'name',
                        header: 'header',
                        purpose: 'purpose',
                        last_post_at: 0,
                        last_root_post_at: 0,
                        creator_id: 'id',
                        scheme_id: 'id',
                        group_constrained: false,
                    }}
                />
            </ComponentWrapper>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test('should not have edit and delete actions if user does not have permissions to change', () => {
        function emptyFunction() {} //eslint-disable-line no-empty-function

        const {container} = render(
            <ComponentWrapper>
                <InstalledIncomingWebhook
                    key={1}
                    incomingWebhook={incomingWebhook}
                    onDelete={emptyFunction}
                    creator={{username: 'creator'}}
                    canChange={false}
                    team={{
                        id: teamId,
                        name: 'test',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        type: 'O',
                        display_name: 'name',
                        scheme_id: 'id',
                        allow_open_invite: false,
                        group_constrained: false,
                        description: '',
                        email: '',
                        company_name: '',
                        allowed_domains: '',
                        invite_id: '',
                    }}
                    channel={{
                        id: '1jiw9kphbjrntfyrm7xpdcya4o',
                        name: 'town-square',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        team_id: teamId,
                        type: 'O',
                        display_name: 'name',
                        header: 'header',
                        purpose: 'purpose',
                        last_post_at: 0,
                        last_root_post_at: 0,
                        creator_id: 'id',
                        scheme_id: 'id',
                        group_constrained: false,
                    }}
                />
            </ComponentWrapper>,
        );
        const element = container.querySelector('.item-actions');
        expect(element).not.toBeInTheDocument();
    });

    test('should have edit and delete actions if user can change webhook', () => {
        function emptyFunction() {} //eslint-disable-line no-empty-function

        render(
            <ComponentWrapper>
                <InstalledIncomingWebhook
                    key={1}
                    incomingWebhook={incomingWebhook}
                    onDelete={emptyFunction}
                    creator={{username: 'creator'}}
                    canChange={true}
                    team={{
                        id: teamId,
                        name: 'test',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        type: 'O',
                        display_name: 'name',
                        scheme_id: 'id',
                        allow_open_invite: false,
                        group_constrained: false,
                        description: '',
                        email: '',
                        company_name: '',
                        allowed_domains: '',
                        invite_id: '',
                    }}
                    channel={{
                        id: '1jiw9kphbjrntfyrm7xpdcya4o',
                        name: 'town-square',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        team_id: teamId,
                        type: 'O',
                        display_name: 'name',
                        header: 'header',
                        purpose: 'purpose',
                        last_post_at: 0,
                        last_root_post_at: 0,
                        creator_id: 'id',
                        scheme_id: 'id',
                        group_constrained: false,
                    }}
                />
            </ComponentWrapper>,
        );
        const editElement = screen.getByText('Edit');
        const deleteElement = screen.getByText('Delete');
        expect(editElement).toBeInTheDocument();
        expect(deleteElement).toBeInTheDocument();
    });

    test('Should have the same name and description on view as it has in incomingWebhook', () => {
        function emptyFunction() {} //eslint-disable-line no-empty-function

        render(
            <ComponentWrapper>
                <InstalledIncomingWebhook
                    key={1}
                    incomingWebhook={incomingWebhook}
                    onDelete={emptyFunction}
                    creator={{username: 'creator'}}
                    canChange={false}
                    team={{
                        id: teamId,
                        name: 'test',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        type: 'O',
                        display_name: 'name',
                        scheme_id: 'id',
                        allow_open_invite: false,
                        group_constrained: false,
                        description: '',
                        email: '',
                        company_name: '',
                        allowed_domains: '',
                        invite_id: '',
                    }}
                    channel={{
                        id: '1jiw9kphbjrntfyrm7xpdcya4o',
                        name: 'town-square',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        team_id: teamId,
                        type: 'O',
                        display_name: 'name',
                        header: 'header',
                        purpose: 'purpose',
                        last_post_at: 0,
                        last_root_post_at: 0,
                        creator_id: 'id',
                        scheme_id: 'id',
                        group_constrained: false,
                    }}
                />
            </ComponentWrapper>,
        );
        const description = screen.getByText('build status');
        const name = screen.getByText('build');

        expect(description).toBeInTheDocument();
        expect(name).toBeInTheDocument();
    });

    test('Should not display description as it is null', () => {
        function emptyFunction() {} //eslint-disable-line no-empty-function
        const newIncomingWebhook: IncomingWebhook = {...incomingWebhook, description: ''};
        const {container} = render(
            <ComponentWrapper>
                <InstalledIncomingWebhook
                    key={1}
                    incomingWebhook={newIncomingWebhook}
                    onDelete={emptyFunction}
                    creator={{username: 'creator'}}
                    canChange={false}
                    team={{
                        id: teamId,
                        name: 'test',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        type: 'O',
                        display_name: 'name',
                        scheme_id: 'id',
                        allow_open_invite: false,
                        group_constrained: false,
                        description: '',
                        email: '',
                        company_name: '',
                        allowed_domains: '',
                        invite_id: '',
                    }}
                    channel={{
                        id: '1jiw9kphbjrntfyrm7xpdcya4o',
                        name: 'town-square',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        team_id: teamId,
                        type: 'O',
                        display_name: 'name',
                        header: 'header',
                        purpose: 'purpose',
                        last_post_at: 0,
                        last_root_post_at: 0,
                        creator_id: 'id',
                        scheme_id: 'id',
                        group_constrained: false,
                    }}
                />
            </ComponentWrapper>,
        );
        expect(container.querySelector('.item-details__description')).not.toBeInTheDocument();
    });

    test('Should not render any nodes as there are no filtered results', () => {
        function emptyFunction() {} //eslint-disable-line no-empty-function
        const {container} = render(
            <ComponentWrapper>
                <InstalledIncomingWebhook
                    key={1}
                    incomingWebhook={incomingWebhook}
                    onDelete={emptyFunction}
                    creator={{username: 'creator'}}
                    filter={'someLongText'}
                    canChange={false}
                    team={{
                        id: teamId,
                        name: 'test',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        type: 'O',
                        display_name: 'name',
                        scheme_id: 'id',
                        allow_open_invite: false,
                        group_constrained: false,
                        description: '',
                        email: '',
                        company_name: '',
                        allowed_domains: '',
                        invite_id: '',
                    }}
                    channel={{
                        id: '1jiw9kphbjrntfyrm7xpdcya4o',
                        name: 'town-square',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        team_id: teamId,
                        type: 'O',
                        display_name: 'name',
                        header: 'header',
                        purpose: 'purpose',
                        last_post_at: 0,
                        last_root_post_at: 0,
                        creator_id: 'id',
                        scheme_id: 'id',
                        group_constrained: false,
                    }}
                />
            </ComponentWrapper>,
        );
        expect(container).toBeEmptyDOMElement();
    });

    test('Should render a webhook item as filtered result is true', () => {
        function emptyFunction() {} //eslint-disable-line no-empty-function
        const {container} = render(
            <ComponentWrapper>
                <InstalledIncomingWebhook
                    key={1}
                    incomingWebhook={incomingWebhook}
                    onDelete={emptyFunction}
                    creator={{username: 'creator'}}
                    filter={'buil'}
                    canChange={true}
                    team={{
                        id: teamId,
                        name: 'test',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        type: 'O',
                        display_name: 'name',
                        scheme_id: 'id',
                        allow_open_invite: false,
                        group_constrained: false,
                        description: '',
                        email: '',
                        company_name: '',
                        allowed_domains: '',
                        invite_id: '',
                    }}
                    channel={{
                        id: '1jiw9kphbjrntfyrm7xpdcya4o',
                        name: 'town-square',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        team_id: teamId,
                        type: 'O',
                        display_name: 'name',
                        header: 'header',
                        purpose: 'purpose',
                        last_post_at: 0,
                        last_root_post_at: 0,
                        creator_id: 'id',
                        scheme_id: 'id',
                        group_constrained: false,
                    }}
                />
            </ComponentWrapper>,
        );
        expect(container.querySelector('.item-details')).toBeInTheDocument();
    });

    test('Webhooks should have the associated channel', () => {
        function emptyFunction() {} //eslint-disable-line no-empty-function
        const {container} = render(
            <ComponentWrapper>
                <InstalledIncomingWebhook
                    key={1}
                    incomingWebhook={incomingWebhook}
                    onDelete={emptyFunction}
                    creator={{username: 'creator'}}
                    filter={'buil'}
                    canChange={true}
                    team={{
                        id: teamId,
                        name: 'test',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        type: 'O',
                        display_name: 'name',
                        scheme_id: 'id',
                        allow_open_invite: false,
                        group_constrained: false,
                        description: '',
                        email: '',
                        company_name: '',
                        allowed_domains: '',
                        invite_id: '',
                    }}
                    channel={{
                        id: '1jiw9kphbjrntfyrm7xpdcya4o',
                        name: 'town-square',
                        create_at: 1502455422406,
                        delete_at: 0,
                        update_at: 1502455422406,
                        team_id: teamId,
                        type: 'O',
                        display_name: 'Town Square',
                        header: 'header',
                        purpose: 'purpose',
                        last_post_at: 0,
                        last_root_post_at: 0,
                        creator_id: 'id',
                        scheme_id: 'id',
                        group_constrained: false,
                    }}
                />
            </ComponentWrapper>,
        );
        expect(container.querySelector('.item-details__channel_name')).toBeInTheDocument();
    });

    test('matchesFilter should return true as it allows filter by channel display_name', () => {
        const result = matchesFilter(
            incomingWebhook,
            {
                id: '1jiw9kphbjrntfyrm7xpdcya4o',
                name: 'town-square',
                create_at: 1502455422406,
                delete_at: 0,
                update_at: 1502455422406,
                team_id: teamId,
                type: 'O',
                display_name: 'TS Display Name',
                header: 'header',
                purpose: 'purpose',
                last_post_at: 0,
                last_root_post_at: 0,
                creator_id: 'id',
                scheme_id: 'id',
                group_constrained: false,
            },
            'ts disp',
        );
        expect(result).toBe(true);
    });

    test("matchesFilter should return false, as filter doesn't match with channel display_name", () => {
        const result = matchesFilter(
            incomingWebhook,
            {
                id: '1jiw9kphbjrntfyrm7xpdcya4o',
                name: 'town-square',
                create_at: 1502455422406,
                delete_at: 0,
                update_at: 1502455422406,
                team_id: teamId,
                type: 'O',
                display_name: 'TS Display Name',
                header: 'header',
                purpose: 'purpose',
                last_post_at: 0,
                last_root_post_at: 0,
                creator_id: 'id',
                scheme_id: 'id',
                group_constrained: false,
            },
            'ts disop',
        );
        expect(result).toBe(false);
    });
});
