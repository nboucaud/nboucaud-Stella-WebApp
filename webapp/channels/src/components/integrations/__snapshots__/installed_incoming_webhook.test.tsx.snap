// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/integrations/InstalledIncomingWebhook should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="backstage-list__item"
  >
    <div
      class="item-details"
    >
      <div
        class="item-details__row d-flex flex-column flex-md-row justify-content-between"
      >
        <strong
          class="item-details__name"
        >
          build
        </strong>
        <div
          class="item-actions"
        >
          <a>
            Edit
          </a>
           - 
          <button
            class="color--link style--none"
          >
            Delete
          </button>
        </div>
      </div>
      <div
        class="item-details__row"
      >
        <span
          class="item-details__description"
        >
          build status
        </span>
      </div>
      <div
        class="item-details__row"
      >
        <span
          class="item-details__url word-break--all"
        >
          URL: http://localhost:8065/hooks/9w96t4nhbfdiij64wfqors4i1r
          <a
            class="fa fa-copy ml-2"
            data-testid="copyText"
            href="#"
          />
        </span>
      </div>
      <div
        class="item-details__row"
      >
        <span
          class="item-details__channel_name word-break--all"
        >
          Channel: name
          <a
            class="fa fa-copy ml-2"
            data-testid="copyText"
            href="#"
          />
        </span>
      </div>
      <div
        class="item-details__row"
      >
        <span
          class="item-details__creation"
        >
          Created by creator on Friday, August 11, 2017
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
